test/e2e/all_in_one_test.go

```diff
@@ -642,6 +681,187 @@ func verifyIngress(ctx context.Context, t *testing.T, env environments.Environme
 	}, ingressWait, time.Second)
 }
 
+func deployGateway(ctx context.Context, t *testing.T, env environments.Environment) *gatewayv1alpha2.Gateway {
+	gc, err := gatewayclient.NewForConfig(env.Cluster().Config())
+	assert.NoError(t, err)
```

Comment:
This is a situation where we _can't_ proceed because all forward logic is dependent on this succeeding, so we should use a `require`:

```suggestion
	require.NoError(t, err)
```